open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/src/main/java/org/odpi/openmetadata/accessservices/assetlineage/handlers/ContextHandler.java

```diff
@@ -88,32 +100,16 @@ public AssetContext getAssetContext(String serverName, String userId, String ass
     }
 
 
-    private EntityDetail getEntityDetails(String serverName, String userId, String assetId) throws UserNotAuthorizedException, RepositoryErrorException, EntityProxyOnlyException, InvalidParameterException, EntityNotKnownException, AssetNotFoundException, PropertyServerException {
-        OMRSMetadataCollection metadataCollection = instanceHandler.getMetadataCollection(serverName);
-        EntityDetail entityDetail = metadataCollection.getEntityDetail(userId, assetId);
-
-        if (entityDetail == null) {
-            AssetLineageErrorCode errorCode = AssetLineageErrorCode.ASSET_NOT_FOUND;
-            String errorMessage = errorCode.getErrorMessageId() +
-                    errorCode.getFormattedErrorMessage(assetId, serverName);
-
-            throw new AssetNotFoundException(errorCode.getHttpErrorCode(),
-                    this.getClass().getName(),
-                    "searchForRelationships",
-                    errorMessage,
-                    errorCode.getSystemAction(),
-                    errorCode.getUserAction());
-        }
-
+    private EntityDetail getEntityDetails(String serverName, String userId, String GUID) throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, org.odpi.openmetadata.frameworks.connectors.ffdc.PropertyServerException, org.odpi.openmetadata.frameworks.connectors.ffdc.UserNotAuthorizedException {
+        String methodname = "getEntityDetails";
+        EntityDetail entityDetail = repositoryHandler.getEntityByGUID(userId, GUID, guidParameter, "Any entity type", methodname);
```

Comment:
@mandy-chessell Since this is a generic method to obtain the entitydetail from any entity type, I did not have an expected entity type argument. I expect that this parameter is for logging purposes only, and so I pass "Any entity type" here.